spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/util/HdfsUtils.java

```diff
@@ -48,4 +52,19 @@ public T next() {
   public static <T> Iterable<T> toJavaUtilIterable(final RemoteIterator<T> remoteIterator) {
     return () -> toJavaUtilIterator(remoteIterator);
   }
+
+  public static long computeDirectorySizeInBytes(Path path, Configuration conf) throws IOException {
+    long totalDataSize = 0;
+    FileSystem fs = path.getFileSystem(conf);
```

Comment:
Why the FileSystem is re-created? it will be the same HCFS implementation

Reply:
Using LoadStatistics as discussed.